modules/ocf_dhcp/files/gen-desktop-leases

```diff
@@ -1,11 +1,24 @@
 #!/usr/bin/env python3
+import subprocess
+
 from ocflib.infra.hosts import hosts_by_filter
 
+DESKTOP_LEASES_FILE = '/etc/dhcp/desktop-leases.conf'
+
 desktops = hosts_by_filter('(type=desktop)')
 
-for desktop in desktops:
-    print('host {} {{ hardware ethernet {}; fixed-address {}; }}'.format(
+with open(DESKTOP_LEASES_FILE, 'r') as f:
+    old_contents = sorted([line.strip() for line in f])
+
+new_contents = []
+for desktop in sorted(desktops, key=lambda d: d['cn'][0]):
+    new_contents.append('host {} {{ hardware ethernet {}; fixed-address {}; }}'.format(
         desktop['cn'][0],
         desktop['macAddress'][0],
         desktop['ipHostNumber'][0],
     ))
+
+if new_contents and new_contents != old_contents:
+    with open(DESKTOP_LEASES_FILE, 'w') as f:
+        f.write('\n'.join(new_contents) + '\n')
+    subprocess.check_call(('systemctl', 'restart', 'isc-dhcp-server'))
```

Comment:
What? [You're telling me you don't want to call systemd through DBus?](https://github.com/systemd/systemd/blob/8255430754d3836b5f4713733341d6e2e8bd8b98/src/core/dbus-unit.c#L644)

Kidding.

Reply:
o boi

Reply:
Hey, it's [only a few lines of code](https://stackoverflow.com/questions/33646374/starting-a-systemd-service-via-python) :P

```python
import dbus
sysbus = dbus.SystemBus()
systemd1 = sysbus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1')
manager = dbus.Interface(systemd1, 'org.freedesktop.systemd1.Manager')
job = manager.RestartUnit('isc-dhcp-server.service', 'fail')
```